File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int i=0;
        int j=0;
        int sum=0;
        int minn=INT_MAX;
        while(j<nums.size())
        {
            sum+=nums[j];
            while(sum>=target)
            {
                minn=min(j-i+1,minn);
                sum-=nums[i];
                i++;
            }
            j++;
        }
        return (minn==INT_MAX)?0:minn;
    }
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        i=0
        j=0
        summ=0
        minn=float('inf')
        while(j<len(nums)):
            summ+=nums[j]
            while(summ>=target):
                minn=min(j-i+1,minn)
                summ-=nums[i]
                i+=1
            j+=1
        if(minn==float('inf')):
            return 0
        else:
            return minn